--- Marsop.Ephemeral.Tests/Core/Implementation/FullIntervalTests.cs
+++ Marsop.Ephemeral.Tests/Core/Implementation/FullIntervalTests.cs
@@ -62,16 +62,20 @@
         measure.Should().Be(10);
     }

-    [Fact]
-    public void ToString_ReturnsBaseToStringImplementation()
+    [Theory]
+    [InlineData(true, true, "[2, 8]")]
+    [InlineData(true, false, "[2, 8)")]
+    [InlineData(false, true, "(2, 8]")]
+    [InlineData(false, false, "(2, 8)")]
+    public void ToString_ReturnsBaseToStringImplementation(bool startIncluded, bool endIncluded, string expected)
     {
         // Arrange
-        var interval = new TestFullInterval(2, 8, false, true);
+        var interval = new TestFullInterval(2, 8, startIncluded, endIncluded);

         // Act
         var result = interval.ToString();

         // Assert
-        result.Should().Be("(2, 8]");
+        result.Should().Be(expected);
     }
 }
